main/channel.c

```diff
@@ -712,12 +718,23 @@ static int does_id_conflict(const char *uniqueid)
 {
 	struct ast_channel *conflict;
 	size_t length = 0;
+	struct ast_channel_callback_data callback_data = {.search = uniqueid, .final_match_result = 0};
 
 	if (ast_strlen_zero(uniqueid)) {
 		return 0;
 	}
 
-	conflict = ast_channel_callback(ast_channel_by_uniqueid_cb, (char *) uniqueid, &length, OBJ_NOLOCK);
+  try_again:
```

Comment:
I don't think goto is necessary here, this could be a do while or for (;;) with break

Reply:
Fixed